CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPluginNGTest.java

```diff
@@ -0,0 +1,494 @@
+        final DefaultPluginInteraction interaction = new DefaultPluginInteraction(null, null);
+
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.LEVEL_PARAMETER_ID).setStringValue("Error");
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
+            assertEquals(ex.getNotificationLevel(), PluginNotificationLevel.ERROR);
+            throw ex;
+        }
+    }
+
+    /**
+     * Test of query method, of class TestParametersPlugin. Tests throwing of
+     * fatal pluginException
+     */
+    @Test(expectedExceptions = PluginException.class)
+    public void testQueryException5() throws Exception {
+        System.out.println("throw pluginexception5");
+
+        final TestParametersPlugin instance = new TestParametersPlugin();
+        final PluginParameters result = instance.createParameters();
+
+        final GraphRecordStore recordStore = new GraphRecordStore();
+        final DefaultPluginInteraction interaction = new DefaultPluginInteraction(null, null);
+
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.LEVEL_PARAMETER_ID).setStringValue("Fatal");
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
+            assertEquals(ex.getNotificationLevel(), PluginNotificationLevel.FATAL);
+            throw ex;
+        }
+    }
+
+    /**
+     * Test of query method, of class TestParametersPlugin. Tests not throwing
+     * of pluginException
+     */
+    @Test
+    public void testQueryException6() throws Exception {
+        System.out.println("throw pluginexception6");
+
+        final TestParametersPlugin instance = new TestParametersPlugin();
+        final PluginParameters result = instance.createParameters();
+
+        final GraphRecordStore recordStore = new GraphRecordStore();
+        final DefaultPluginInteraction interaction = new DefaultPluginInteraction(null, null);
+
+        // Set plugin query name here before execution
+        result.getParameters().get(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID).setStringValue("TESTPARAMETERSPLUGIN");
+
+        // Set plugin parameters here before execution
+        result.getParameters().get(TestParametersPlugin.LEVEL_PARAMETER_ID).setStringValue("None");
+
+        try {
+            instance.query(recordStore, interaction, result);
+        } catch (final PluginException ex) {
+            fail();
```

Comment:
I could be wrong but if you're throwing the exception then that should be enough for the unit test to fail without needing this function